nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java

```diff
@@ -0,0 +1,273 @@
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        final MatchingStrategy matchingStrategy = getMatchingStrategy(context);
+
+        cachedMatchingPredicate = null;
+
+        if (matchingStrategy == MatchingStrategy.ENUMERATION
+                && !context.getProperty(ATTRIBUTE_SET).isExpressionLanguagePresent()) {
+            cachedMatchingPredicate = determineMatchingPredicateBasedOnEnumeration(context, null);
+        }
+        if (matchingStrategy == MatchingStrategy.REGEX
+                && !context.getProperty(ATTRIBUTE_REGEX).isExpressionLanguagePresent()) {
+            cachedMatchingPredicate = determineMatchingPredicateBasedOnRegex(context, null);
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final Predicate<String> matchingPredicate = determineMatchingPredicate(context, flowFile);
+
+        final FilterMode filterMode = getFilterMode(context);
+        final Predicate<String> isMatched;
+        if (filterMode == FilterMode.RETAIN) {
+            isMatched = matchingPredicate;
+        } else {
+            isMatched = matchingPredicate.negate();
+        }
+
+        final Set<String> attributesToRemove = new HashSet<>(flowFile.getAttributes().keySet());
+        attributesToRemove.removeIf(isMatched);
+
+        final FlowFile updatedFlowFile = session.removeAllAttributes(flowFile, attributesToRemove);
+        session.transfer(updatedFlowFile, REL_SUCCESS);
+    }
+
+    private Predicate<String> determineMatchingPredicate(ProcessContext context, FlowFile flowFile) {
+        if (cachedMatchingPredicate != null) {
+            return cachedMatchingPredicate;
+        }
+
+        final MatchingStrategy matchingStrategy = getMatchingStrategy(context);
+
+        switch (matchingStrategy) {
+            case ENUMERATION:
+                return determineMatchingPredicateBasedOnEnumeration(context, flowFile);
+            case REGEX:
+                return determineMatchingPredicateBasedOnRegex(context, flowFile);
+            default:
+                throw new IllegalArgumentException(
+                        "Cannot determine matching predicate for unsupported strategy " + matchingStrategy + " !"
+                );
+        }
+    }
+
+    private static Predicate<String> determineMatchingPredicateBasedOnEnumeration(ProcessContext context, FlowFile flowFile) {
+        final String attributeSetDeclaration = getAttributeSet(context, flowFile);
+        final String delimiter = getDelimiter();
+
+        Set<String> attributeSet = Arrays.stream(attributeSetDeclaration.split(Pattern.quote(delimiter)))
+                .map(String::trim)
+                .filter(attributeName -> !attributeName.trim().isEmpty())
```

Comment:
Is it possible to use the `not()` Predicate matcher instead of the `!` in this case?